internal/post/sender.go

```diff
@@ -45,8 +46,11 @@ func (sender *SMTPSender) SendEmail(ctx context.Context, msg *Message) (err erro
 		return err
 	}
 	// close underlying connection
+	// if any unexpected error occurred
 	defer func() {
-		err = errs.Combine(err, client.Close())
+		if err != nil {
```

Comment:
Currently when an error happens earlier then the connection won't be closed.

Reply:
It feels like it always should call `client.Close`, but depending on the outer error might ignore it.

Reply:
@egonelbre  if there is no error that means we called `client.Quit` successfully, which closes the connection. If error some error occurs during sending process then we close it manually in defer

Reply:
could you please describe a situation when it won't be closed?

Reply:
Let's say msg.Bytes() returns a parsing error.

Reply:
@egonelbre so then we return this `err`, defer is called with non-nil `err` and we close the connection, right? May be I am not quite following you

Reply:
if the error is non-nil it doesn't mean the underlying connection is closed